classes/Tools.php

```diff
@@ -1401,45 +1402,13 @@ public static function link_rewrite($str, $utf8_decode = null)
      */
```

Comment:
Do you think we should add @deprecated?

Reply:
We may do this later, but for now it is mostly code factorisation, moving code from legacy to it's tested version in the core.

Reply:
So there will be another PR after to replace all the current usages of `Tools::str2url` right? And that's when we should deprecate the method I guess

Reply:
I think this is too soon to deprecate any of this functions. Before deprecating we would have to replace all usages by copying  the following code into all the current function usages.

```php
$str = (new StringModifier())->str2url((string) $str, (bool) Configuration::get('PS_ALLOW_ACCENTED_CHARS_URL'));
```

This make the code less readable IMO. I think we must wait for the symfony container to be available in both frontend and backend environments, so we can replace usage by just a call to the container (as there is a dependency on the Configuration). And then deprecate the call.